internal/provider/egress_private_endpoint_resource.go

```diff
@@ -0,0 +1,504 @@
+		}
+		*createdEndpoint = *endpoint
+		return nil
+	}
+}
+
+func waitForEgressEndpointCreatedFunc(
+	ctx context.Context,
+	cl client.Service,
+	clusterID string,
+	endpointID string,
+	endpoint *client.EgressPrivateEndpoint,
+) retry.RetryFunc {
+	return func() *retry.RetryError {
+		ep, err := getEgressPrivateEndpoint(ctx, cl, clusterID, endpointID)
+		if err != nil {
+			return retry.RetryableError(
+				fmt.Errorf("encountered an error while getting egress private endpoint, retrying: %s", err.Error()),
+			)
+		} else if ep == nil {
+			return retry.NonRetryableError(fmt.Errorf("egress private endpoint not found"))
+		}
+
+		if !isEgressPrivateEndpointCreatedInCloud(ep) {
+			return retry.RetryableError(fmt.Errorf("egress private endpoint has not been created yet"))
+		}
+
+		*endpoint = *ep
+		return nil
+	}
+}
+
+func isEgressPrivateEndpointCreatedInCloud(endpoint *client.EgressPrivateEndpoint) bool {
+	return endpoint.EndpointConnectionId != "" && endpoint.EndpointAddress != ""
+}
+
+func loadEgressPrivateEndpointIntoTerraformState(endpoint *client.EgressPrivateEndpoint, clusterID string, state *EgressPrivateEndpoint) {
+	state.ID = types.StringValue(endpoint.GetId())
+	state.ClusterID = types.StringValue(clusterID)
+	state.EndpointConnectionID = types.StringValue(endpoint.GetEndpointConnectionId())
+	state.Region = types.StringValue(endpoint.GetRegion())
+	state.TargetServiceIdentifier = types.StringValue(endpoint.GetTargetServiceIdentifier())
+	state.TargetServiceType = types.StringValue(string(endpoint.GetTargetServiceType()))
+	state.EndpointAddress = types.StringValue(endpoint.GetEndpointAddress())
+	state.State = types.StringValue(string(endpoint.GetState()))
+}
+
+func deleteEgressEndpointFunc(ctx context.Context, cl client.Service, clusterID string, endpointID string) retry.RetryFunc {
+	return func() *retry.RetryError {
+		traceAPICall("DeleteEgressPrivateEndpoint")
+		httpResp, err := cl.DeleteEgressPrivateEndpoint(
+			ctx,
+			clusterID,
+			endpointID,
+		)
+		if err != nil {
+			apiErrMsg := formatAPIErrorMessage(err)
+			if httpResp != nil && httpResp.StatusCode == http.StatusNotFound {
+				// Already deleted
+				return nil
+			} else if httpResp != nil && httpResp.StatusCode == http.StatusServiceUnavailable ||
+				strings.Contains(apiErrMsg, "lock") {
+				// Cluster is locked, retry
+				return retry.RetryableError(
+					fmt.Errorf("cluster is locked, retrying: %s", formatAPIErrorMessage(err)),
+				)
+			}
+			return retry.NonRetryableError(err)
+		}
+		return nil
+	}
+}
+
+func waitForEgressEndpointDeletedFunc(
+	ctx context.Context,
+	cl client.Service,
+	clusterID string,
+	endpointID string,
+) retry.RetryFunc {
+	return func() *retry.RetryError {
+		endpoint, err := getEgressPrivateEndpoint(ctx, cl, clusterID, endpointID)
+		if err != nil {
+			return retry.RetryableError(
+				fmt.Errorf("encountered an error while getting egress private endpoint, retrying: %s", err.Error()),
+			)
+		}
```

Comment:
sorry, I must have missed this before, usually we stop retrying on client errors in general with some exceptions (like removing resources if not found, or retrying if locked, etc).  It looks like getEgressPrivateEndpoints is returning standard errors for any client error (except not found) and those will end up being retried.  For example, if the service account didn't have access to the cluster we would return a 403 and it would continue to be retried.